WordPress/wordpress.org
# Forums: Homepage redesign

Try implementing the redesign using the current non-block support theme.

Closes https://meta.trac.wordpress.org/ticket/4007

Rearranges the forums homepage content to match the [design](https://www.figma.com/file/3eK690rwPubLvTEvJIJKRO/Forums?type=design&node-id=902-1140&mode=dev) and applies layout and typography changes to match the other 'Learn' sites ([Dev resources](https://developer.wordpress.org/), [Documentation](https://wordpress.org/documentation/)). ~I've stopped short of styling the whole page as I think there's enough work here to judge whether it's a decent approach.~

~The single forum page in the screenshots has had 0 attention, I've just included it to show the different header layout, font and page width changes. Content widths still need working out; we could do as much or as little here as we want as a first iteration.~

Mostly we can use block theme things:

- theme.json settings are output on frontend and vars and helper classes can be used for styling
- we can register block patterns and render with `do_blocks`
- local nav and search blocks work fine, I expect we can use sidebar containers and other things too

Things that need work:

- [x] Better theme.json integration. I've copied in the [wporg-parent-2021 theme.json](https://github.com/WordPress/wporg-parent-2021/blob/trunk/source/wp-content/themes/wporg-parent-2021/theme.json), and then made some typography modifications to match the Developer site. This is going to be a maintenance burden. Ideally we would load the parent theme.json (rather than duplicating it) and merge with local overrides like we do for other themes. Not sure how to tackle this.
- [x] Better parent theme styles and block styles integration. Loading from a remote url rather than having this theme be a dependant child isn't ideal. Not sure how to tackle this.
- [x] Remap CSS vars
- [x] Button colors
- [x] Profile restyle
- [x] Forum restyle
- [x] Topic restyle
- [x] Content widths, incl responsive
- [x] Fix 'Also viewing' bar fixed position when scrolled
- [x] Topic table headers and row link styles
- [x] Content spacing

### Testing

Needs to be tested in a sandbox.

General things to check for:
- The old WordPress blue shouldn't be used anywhere except parts of the User Account settings. We need to do work in the 2FA repo to update these
- Fonts should be Inter or EB Garamond. Font sizes have had some updates (headings, paragraphs, sidebar), but there is still lot of bbPress text which will need resizing in another iteration.
- Spacings should be generally inline with the new design, eg. content widths, edge spacing, block spacing

#### How to test

Browse the site checking the following pages and functionality:
- Forums home (check links, responsiveness, a11y)
- Forum archive, eg. https://wordpress.org/support/forum/installation/
- Topic
- All Topics
- Profile and sub pages
- Plugin forum, eg. https://wordpress.org/support/plugin/gutenberg/
- Theme forum, eg. https://wordpress.org/support/theme/twentytwentyfour/
- Check that the new Search field below the local nav behaves the same as in prod on each page.
- Moderation tools (nothing should have changed here)
- Local nav

### Screenshots

#### Forums home

| Before | Desktop | Tablet | Mobile |
|-|-|-|-|
| ![wordpress org_support_forums_(3 1536x864) (3)](https://github.com/WordPress/wordpress.org/assets/1017872/af73f984-276c-4418-bef7-6b73d1d0852f) | ![wordpress org_support_forums_(3 1536x864) (5)](https://github.com/WordPress/wordpress.org/assets/1017872/ea86250f-a1c2-4981-b02e-b9468bc29f22) | ![wordpress org_support_forums_(iPad) (2)](https://github.com/WordPress/wordpress.org/assets/1017872/e4403f9d-1eba-43be-8406-114c09c2eff6) | ![wordpress org_support_forums_(Samsung Galaxy S20 Ultra) (2)](https://github.com/WordPress/wordpress.org/assets/1017872/614cbc78-e959-40b6-a0d0-f0b38d0ae747) |


#### Forum list

| Before | Desktop | Tablet | Mobile |
|-|-|-|-|
| ![wordpress org_support_forum_installation_(3 1536x864) (2)](https://github.com/WordPress/wordpress.org/assets/1017872/15fe0055-5ab7-41f7-a7bf-2cdb20ee65d0) | ![wordpress org_support_forum_installation_(3 1536x864) (4)](https://github.com/WordPress/wordpress.org/assets/1017872/8f077f79-6a5f-4533-8b43-e897d0a816d0) | ![wordpress org_support_forum_installation_(iPad) (2)](https://github.com/WordPress/wordpress.org/assets/1017872/48150390-2acb-4e33-b4cd-326c150e62e4) | ![wordpress org_support_forum_installation_(Samsung Galaxy S20 Ultra) (2)](https://github.com/WordPress/wordpress.org/assets/1017872/38e53413-f7ad-4d9c-a5c5-4cdebbdb3ba0) |

#### Topic

| Before | Desktop | Tablet | Mobile |
|-|-|-|-|
| ![wordpress org_support_topic_is-htaccess-working-environment-required-for-wordpress_(3 1536x864)](https://github.com/WordPress/wordpress.org/assets/1017872/f43ca1b3-c7a5-4542-9a6e-5588e09ccd1f) | ![wordpress org_support_topic_is-htaccess-working-environment-required-for-wordpress_(3 1536x864) (2)](https://github.com/WordPress/wordpress.org/assets/1017872/6a419329-3dc6-4b21-bb91-d457d930f7e7) | ![wordpress org_support_topic_is-htaccess-working-environment-required-for-wordpress_(iPad) (1)](https://github.com/WordPress/wordpress.org/assets/1017872/7029b649-5423-44ba-abd7-e441d6da3a41) | ![wordpress org_support_topic_is-htaccess-working-environment-required-for-wordpress_(Samsung Galaxy S20 Ultra) (1)](https://github.com/WordPress/wordpress.org/assets/1017872/a47a2525-7dd6-4e68-bc14-9a947217a44a) |

#### Welcome

| Before | Desktop | Tablet | Mobile |
|-|-|-|-|
| ![wordpress org_support_welcome_(3 1536x864)](https://github.com/WordPress/wordpress.org/assets/1017872/0ad6f873-e2d8-49c4-96fa-198754f3cf9e) | ![wordpress org_support_welcome_(3 1536x864) (1)](https://github.com/WordPress/wordpress.org/assets/1017872/dd316177-bb1a-4da4-9927-865142096fbd) | ![wordpress org_support_welcome_(iPad)](https://github.com/WordPress/wordpress.org/assets/1017872/603b9473-f9be-455a-b288-79fb07fd3f8e) | ![wordpress org_support_welcome_(Samsung Galaxy S20 Ultra)](https://github.com/WordPress/wordpress.org/assets/1017872/6af6d928-4394-45f4-b39b-be1739bfe0cb) |

#### Profile

| Before | Desktop | Tablet | Mobile |
|-|-|-|-|
| ![wordpress org_support_users_adamwood_(3 1536x864)](https://github.com/WordPress/wordpress.org/assets/1017872/ea29dfe1-5a44-42c2-bebc-0010c0ab3e8f) | ![wordpress org_support_users_adamwood_(3 1536x864) (1)](https://github.com/WordPress/wordpress.org/assets/1017872/d5f6f473-2d9e-419b-8903-98a07b1dab2f) | ![wordpress org_support_users_adamwood_(iPad)](https://github.com/WordPress/wordpress.org/assets/1017872/8ff388df-2618-4c35-b040-af38fc2598ac) | ![wordpress org_support_users_adamwood_(Samsung Galaxy S20 Ultra)](https://github.com/WordPress/wordpress.org/assets/1017872/505bb4de-4d1a-4d76-b53c-9cb05cd413ee) |

#### Account

| Before | Desktop | Tablet | Mobile |
|-|-|-|-|
| ![wordpress org_support_users_adamwood_edit_account_(3 1536x864)](https://github.com/WordPress/wordpress.org/assets/1017872/ca7c6721-9dab-40e9-b54d-369090685d04) | ![wordpress org_support_users_adamwood_edit_account_(3 1536x864) (1)](https://github.com/WordPress/wordpress.org/assets/1017872/f1f279f2-be82-40af-ba50-de65a9ebf70b) | ![wordpress org_support_users_adamwood_edit_account_(iPad)](https://github.com/WordPress/wordpress.org/assets/1017872/8121056a-cbea-4a4c-8b25-367890a40b7d) | ![wordpress org_support_users_adamwood_edit_account_(Samsung Galaxy S20 Ultra)](https://github.com/WordPress/wordpress.org/assets/1017872/de43e911-4654-4648-ae26-f714615ba134) |
